packages/core/content-type-builder/admin/src/utils/getMaxDepth.ts

```diff
@@ -0,0 +1,95 @@
+import type { ComponentWithChildren } from '../components/DataManagerProvider/utils/retrieveComponentsThatHaveComponents';
+import type { NestedComponent } from '../components/DataManagerProvider/utils/retrieveNestedComponents';
+import type { Internal } from '@strapi/types';
+
+const findComponent = <T extends { component: Internal.UID.Component }>(
+  componentName: Internal.UID.Schema,
+  components: Array<T>
+) => {
+  return components.find((c) => c.component === componentName);
```

Comment:
Could there be multiple components matching the UID in this array?

Reply:
nah it's unique, only one will exist